discovery_engine/lib/src/ffi/types/duration.dart

```diff
@@ -33,18 +32,71 @@ extension DurationFfi on Duration {
   /// Sub microseconds precision will be ignored.
   ///
   /// In case of a too large durations an exception will be throw.
-  static Duration readNative(final Pointer<RustDuration> durationPlace) {
+  static Duration readNative(final Pointer<RustDuration> place) =>
+      _Helper.readWithReaders(
+        ffi.get_duration_seconds,
+        ffi.get_duration_nanos,
+        place,
+      );
+}
+
+extension OptionDurationFfi on Duration? {
+  void writeNative(final Pointer<RustOptionDuration> place) {
+    final self = this;
+    if (self == null) {
+      ffi.init_option_duration_none_at(place);
+    } else {
+      self.writeWithWriter(ffi.init_option_duration_some_at, place);
+    }
+  }
+
+  /// Reads a dart [Duration] from an optional rust duration.
+  ///
+  /// Be aware that darts [Duration] has both less precision and  is more
+  /// limited wrt. the max duration as it stores the duration as an `int`
+  /// of microseconds.
+  ///
+  /// Sub microseconds precision will be ignored.
+  ///
+  /// In case of a too large durations an exception will be throw.
```

Comment:
```suggestion
  /// Be aware that dart's [Duration] has both less precision and is more
  /// limited wrt. the max duration as it stores the duration as an `int`
  /// of microseconds.
  ///
  /// Sub microsecond precision will be ignored.
  ///
  /// In case of too large durations an exception will be thrown.
```